drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopupView.java

```diff
@@ -0,0 +1,472 @@
+                GuidedDecisionTableImageResources508.INSTANCE.EditDisabled(),
+                GuidedDecisionTableConstants.INSTANCE.EditTheOperatorThatIsUsedToCompareDataWithThisField() );
+
+        entryPointName = new TextBox();
+        header = new TextBox();
+        valueListWidget = new TextBox();
+    }
+
+    public void initializeView() {
+        HorizontalPanel patternWidget = new HorizontalPanel();
+        patternWidget.add( patternLabel );
+
+        //Pattern selector
+        changePattern.addClickHandler(
+                new ClickHandler() {
+                    public void onClick( ClickEvent w ) {
+                        presenter.showChangePattern( w );
+                    }
+                }
+        );
+        changePattern.setEnabled( !this.presenter.isReadOnly() );
+        patternWidget.add( changePattern );
+
+        addAttribute( GuidedDecisionTableConstants.INSTANCE.Pattern(),
+                patternWidget );
+
+        //Radio buttons for Calculation Type
+        switch ( this.presenter.getTableFormat() ) {
+            case EXTENDED_ENTRY:
+                HorizontalPanel valueTypes = new HorizontalPanel();
+                valueTypes.add( literal );
+                valueTypes.add( formula );
+                valueTypes.add( predicate );
+                addAttribute( new StringBuilder( GuidedDecisionTableConstants.INSTANCE.CalculationType() ).append( GuidedDecisionTableConstants.COLON ).toString(),
+                        valueTypes );
+
+                switch ( this.presenter.getConstraintValueType() ) {
+                    case BaseSingleFieldConstraint.TYPE_LITERAL:
+                        literal.setValue( true );
+                        break;
+                    case BaseSingleFieldConstraint.TYPE_RET_VALUE:
+                        formula.setValue( true );
+                        break;
+                    case BaseSingleFieldConstraint.TYPE_PREDICATE:
+                        predicate.setValue( true );
+                }
+
+                if ( !this.presenter.isReadOnly() ) {
+                    literal.addClickHandler( new ClickHandler() {
+                        public void onClick( ClickEvent w ) {
+                            presenter.setFactField( null );
+                            presenter.applyConsTypeChange( BaseSingleFieldConstraint.TYPE_LITERAL );
+                        }
+                    } );
+                }
+
+                if ( !this.presenter.isReadOnly() ) {
+                    formula.addClickHandler( new ClickHandler() {
+                        public void onClick( ClickEvent w ) {
+                            presenter.setFactField( null );
+                            presenter.applyConsTypeChange( BaseSingleFieldConstraint.TYPE_RET_VALUE );
+                        }
+                    } );
+                }
+
+                if ( !this.presenter.isReadOnly() ) {
```

Comment:
IMO, I would keep it simple with only one `if ( !this.presenter.isReadOnly() )`. But, it's just a personal option :-)